sql-jdbc/src/main/java/org/opensearch/jdbc/types/TimeType.java

```diff
@@ -24,41 +28,68 @@ public Time fromValue(Object value, Map<String, Object> conversionParams) throws
     if (value == null) {
       return null;
     }
+    Calendar calendar = conversionParams != null ? (Calendar) conversionParams.get("calendar") : null;
     if (value instanceof Time) {
-      return asTime((Time) value);
+      return (Time) value;
     } else if (value instanceof String) {
-      return asTime((String) value);
+      return asTime((String) value, calendar);
     } else if (value instanceof Number) {
       return this.asTime((Number) value);
     } else {
       throw objectConversionException(value);
     }
   }
 
-  public Time asTime(Time value) {
-    return localTimetoSqlTime(value.toLocalTime());
-  }
+  public Time asTime(String value, Calendar calendar) throws SQLException {
+    try {
+      // Make some effort to understand ISO format
+      if (value.length() > 11 && value.charAt(10) == 'T') {
```

Comment:
Are you getting this from a different file?  Is there a java library that will do this already?

Reply:
Yes, the logic is the same as in `sql-jdbc/src/main/java/org/opensearch/jdbc/types/TimestampType.java`